classes/query/Repository.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+/**
+ * @file classes/query/Repository.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Repository
+ *
+ * @see Query
+ *
+ * @brief Operations for retrieving and modifying Query objects.
+ */
+
+namespace PKP\query;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\notification\NotificationManager;
+use APP\submission\Submission;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Mail;
+use PKP\db\DAORegistry;
+use PKP\mail\Mailable;
+use PKP\note\Note;
+use PKP\notification\Notification;
+use PKP\notification\NotificationSubscriptionSettingsDAO;
+use PKP\security\Role;
+use PKP\stageAssignment\StageAssignment;
+use PKP\user\User;
+
+class Repository
+{
+    /**
+     * Fetch a query by symbolic info, building it if needed.
+     */
+    public function build(int $assocType, int $assocId, int $userId, int $stageId, float $seq = 0, bool $closed = false): Query
+    {
+        return Query::withUserId($userId)
+            ->withAssoc($assocType, $assocId)
+            ->firstOr(fn() => Query::create([
+                'assocType' => $assocType,
+                'assocId' => $assocId,
+                'userId' => $userId,
+                'stageId' => $stageId,
+                'seq' => $seq,
+                'closed' => $closed
+            ]));
+    }
+
+    /**
+     * Retrieve a count of all open queries totalled by stage
+     *
+     * @param int[] $participantIds Only include queries with these participants
+     *
+     * @return array<int,int> [int $stageId => int $count]
+     */
+    public function countOpenPerStage(int $submissionId, ?array $participantIds = null)
```

Comment:
Still in progress - this is only currently used in `lib/pkp/classes/submission/maps/Schema.php`